model/flow/cluster.go

```diff
@@ -60,15 +62,30 @@ func NewClusterList(assignments AssignmentList, collectors IdentityList) (Cluste
 
 	// replicate the identifier list but use identities instead
 	clusters := make(ClusterList, 0, len(assignments))
-	for _, participants := range assignments {
+	for i, participants := range assignments {
 		cluster := make(IdentityList, 0, len(participants))
-		for _, participantID := range participants {
+		if len(participants) == 0 {
+			return nil, fmt.Errorf("particpants in assignment list is empty, cluster index %v", i)
+		}
+
+		// Check assignments is sorted in canonical order
+		prev := participants[0]
+
+		for i, participantID := range participants {
 			participant, found := lookup[participantID]
 			if !found {
 				return nil, fmt.Errorf("could not find collector identity (%x)", participantID)
 			}
 			cluster = append(cluster, participant)
 			delete(lookup, participantID)
+
+			if i > 0 {
+				if bytes.Compare(prev[:], participantID[:]) > 0 {
```

Comment:
```suggestion
				if !IdentifierCanonical(prev, participantID) {
```
Directly use the canonical ordering function, otherwise this might break if the canonical ordering changed.

Reply:
Another suggestion:

Now that we have canonical ordering defined for both `[]Identity` and `[]Identifier`, we could make some changes to keep them consistent:
* define the `Identity` ordering in terms of the `Identifier` ordering
```go
// model/flow/order/identity.go
func Canonical(identity1 *flow.Identity, identity2 *flow.Identity) bool {
	return IdentifierCanonical(identity1.NodeID, identity2.NodeID)
}
```
* add a test case to make sure the canonical orderings match
```go
identities := unittest.IdentityListFixture()
assert.Equal(t, identities.Sort(Canonical).NodeIDs(), identities.NodeIDs().Sort(IdentifierCanonical))
```